p2p/client.go

```diff
@@ -189,27 +238,14 @@ func (s *peer) GetSignaturesForDocument(ctx context.Context, identityService ide
 	}
 
 	var count int
+	peerCtx, _ := context.WithTimeout(ctx, nc.GetP2PConnectionTimeout())
 	for _, collaborator := range extCollaborators {
 		collaboratorID, err := identity.ToCentID(collaborator)
 		if err != nil {
 			return centerrors.Wrap(err, "failed to convert to CentID")
 		}
-		id, err := identityService.LookupIdentityForID(collaboratorID)
```

Comment:
These two expensive calls happen inside collaborator specific go routines now because of your comment @mikiquantum :)